### PR TITLE
drm_common: skip cards that don't have connected outputs

### PR DESCRIPTION
**Pull request by llyyr**
It's possible for systems to have multiple cards, and the first capable card to not have a connected output. Skip such cards and continue iterating until we find one with a connected output.

### CODE REVIEW

File: video/out/drm_common.c [hunk 1]
```diff
@@ -816,6 +816,19 @@ static bool card_supports_kms(const char *path)
     return ret;
 }
 
+static bool card_has_connection(const char *path)
+{
+    int fd = open_card_path(path);
+    bool ret = true;
+    if (fd != -1) {
+        drmModeRes *res = drmModeGetResources(fd);
+        ret = get_first_connected_connector(res, fd);
```
Comment (sfan5) [inline video/out/drm_common.c:825]: this needs to be freed

Comment (llyyr) [reply video/out/drm_common.c]: Oops, fixed it

File: video/out/drm_common.c [hunk 2]
```diff
@@ -816,6 +816,22 @@ static bool card_supports_kms(const char *path)
...
 }
 
+static bool card_has_connection(const char *path)
+{
+    int fd = open_card_path(path);
+    bool ret = false;
+    if (fd != -1) {
+        drmModeRes *res = drmModeGetResources(fd);
+        drmModeConnector *connector = get_first_connected_connector(res, fd);
+        if (connector && connector->connection == DRM_MODE_CONNECTED)
```
Comment (sfan5) [inline video/out/drm_common.c:826]: `get_first_connected_connector` already checks this

File: video/out/drm_common.c [hunk 3]
```diff
@@ -816,6 +816,22 @@ static bool card_supports_kms(const char *path)
     return ret;
 }
 
+static bool card_has_connection(const char *path)
+{
+    int fd = open_card_path(path);
+    bool ret = false;
+    if (fd != -1) {
+        drmModeRes *res = drmModeGetResources(fd);
+        drmModeConnector *connector = get_first_connected_connector(res, fd);
```
Comment (sfan5) [inline video/out/drm_common.c:825]: if it's possible for `res` to be null that needs to be checked here

**Review by sfan5 — APPROVED** [review]
works.

**Pull request merged**